Components/components.js

```diff
@@ -0,0 +1,93 @@
+//alert close example
+
+const alertExampleBox = document.querySelector("#alert-example-close");
+const alertExampleCloseBtn = document.querySelector("#alert-example-close-btn");
+
+alertExampleCloseBtn.addEventListener("click",()=>{
+    alertExampleBox.style.display = "none";
+})
+
+//form validation example
+
+const allForms = document.querySelectorAll(".submit-form-example");
+
+allForms.forEach((form)=>{
+    form.addEventListener("submit", (e)=>{
+        if(!form.checkValidity()){
+            e.preventDefault();
+        }
+        form.classList.add("form-validated");
+        
+    }, false)
+})
+
+//Modal
+
+const modalCTA = document.querySelector("#example-modal-CTA");
+const modalInterstitial = document.querySelector("#example-modal-interstitial");
+const modalCloseBtn = document.querySelector("#example-modal-close");
+const modalBackBtn = document.querySelector("#example-modal-back");
+
+modalCTA.addEventListener("click", ()=>{
+    modalInterstitial.classList.add("active");
+    document.body.style.overflow = "hidden"
+})
+
+modalCloseBtn.addEventListener("click", ()=>{
+    modalInterstitial.classList.remove("active");
+    document.body.style.overflow = "visible";
+})
+
+modalBackBtn.addEventListener("click", ()=>{
+    modalInterstitial.classList.remove("active");
+    document.body.style.overflow = "visible";
+})
```

Comment:
Since both the buttons are implementing the same functionality, it could be extracted out as a separate function and passed as a callback to these addEventListeners.

Reply:
Sure. I will update this.